07_memory/usr_mem_test.log

```diff
@@ -0,0 +1,370 @@
+| Id: 8
+| alloc time: 401(ns)	free time: 206(ns)
+| Id: 9
+| alloc time: 1079(ns)	free time: 609(ns)
+| Id: 10
+| alloc time: 2285(ns)	free time: 171(ns)
+| Id: 11
+| alloc time: 656(ns)	free time: 476(ns)
+| Id: 12
+| alloc time: 1082(ns)	free time: 165(ns)
+| Id: 13
+| alloc time: 1131(ns)	free time: 119(ns)
+| Id: 14
+| alloc time: 2438(ns)	free time: 162(ns)
+| Id: 15
+| alloc time: 4552(ns)	free time: 156(ns)
+| Id: 16
+| alloc time: 8231(ns)	free time: 159(ns)
+| Id: 17
+| alloc time: 17115(ns)	free time: 165(ns)
+| Id: 18
+| alloc time: 32796(ns)	free time: 241(ns)
+| Id: 19
+| alloc time: 68147(ns)	free time: 249(ns)
+| Id: 20
+| alloc time: 131245(ns)	free time: 159(ns)
+| Id: 21
+| alloc time: 246398(ns)	free time: 556(ns)
+| Id: 22
+| alloc time: 543712(ns)	free time: 514(ns)
+| Id: 23
+| alloc time: 1215428(ns)	free time: 797(ns)
+| Id: 24
+| alloc time: 12450360(ns)	free time: 1176(ns)
+| Id: 25
+| alloc time: 38038(ns)	free time: 21495(ns)
+| Id: 26
+| alloc time: 12967(ns)	free time: 13839(ns)
+| Id: 27
+| alloc time: 9896(ns)	free time: 12671(ns)
+| Id: 28
+| alloc time: 10052(ns)	free time: 14967(ns)
+| Id: 29
+| alloc time: 10453(ns)	free time: 19814(ns)
+| Id: 30
+| alloc time: 12699(ns)	free time: 25713(ns)
+| Id: 31
+| alloc time: 12666(ns)	free time: 29800(ns)
+| Id: 32
+| alloc time: 12315(ns)	free time: 28432(ns)
+| Id: 33
+| alloc time: 11631(ns)	free time: 15857(ns)
+| Id: 34
+| alloc time: 10809(ns)	free time: 15154(ns)
+| Id: 35
+| ### Memory cannot be allocated!
+| Average alloc time: 232375(ns)	average free time: 3209(ns)
+====== Run memory test allocation! ============================
+|
+| function call: alloca
+| test case:     2^x + 1
+|
+===============================================================
+| Id: 0
+| alloc time: 197(ns)	free time: 0(ns)
+| Id: 1
+| alloc time: 119(ns)	free time: 0(ns)
+| Id: 2
+| alloc time: 76(ns)	free time: 0(ns)
+| Id: 3
+| alloc time: 77(ns)	free time: 0(ns)
+| Id: 4
+| alloc time: 84(ns)	free time: 0(ns)
+| Id: 5
+| alloc time: 80(ns)	free time: 0(ns)
+| Id: 6
+| alloc time: 85(ns)	free time: 0(ns)
+| Id: 7
+| alloc time: 97(ns)	free time: 0(ns)
+| Id: 8
+| alloc time: 74(ns)	free time: 0(ns)
+| Id: 9
+| alloc time: 86(ns)	free time: 0(ns)
+| Id: 10
+| alloc time: 76(ns)	free time: 0(ns)
+| Id: 11
+| alloc time: 77(ns)	free time: 0(ns)
+| Id: 12
+| alloc time: 384(ns)	free time: 0(ns)
+| Id: 13
+| alloc time: 5225(ns)	free time: 0(ns)
+| Id: 14
+| alloc time: 7355(ns)	free time: 0(ns)
+| Id: 15
+| alloc time: 14132(ns)	free time: 0(ns)
+| Id: 16
+| alloc time: 28474(ns)	free time: 0(ns)
+| Id: 17
+| alloc time: 62657(ns)	free time: 0(ns)
+| Id: 18
+| alloc time: 117659(ns)	free time: 0(ns)
+| Id: 19
+| alloc time: 236317(ns)	free time: 0(ns)
+| Id: 20
+| alloc time: 440248(ns)	free time: 0(ns)
+| Id: 21
+| alloc time: 899967(ns)	free time: 0(ns)
+| Id: 22
+| alloc time: 1781597(ns)	free time: 0(ns)
+| Id: 23
+Segmentation fault (core dumped)
```

Comment:
Is it possible to omit this segmentation fault?

Reply:
Yes, just add signal handler. But it's bad practice I guess.